src/utils/WorkspacesUtils.js

```diff
@@ -55,6 +67,69 @@ const patchWorkspaceWithDatasetManagerConfiguration = (workspace) => {
   }
 };
 
+const checkDatasetManagerConfiguration = (workspace) => {
+  const logWarning = (warning) => console.warn(`Dataset manager configuration: ${warning}`);
+  const config = workspace?.webApp?.options?.datasetManager;
+  if (config == null) return;
+  if (!(config instanceof Object)) {
+    logWarning('"datasetManager" must be an Object');
+    return;
+  }
+
+  const { graphIndicators, categories, queries } = config;
+
+  if (queries != null && !(queries instanceof Array)) {
+    logWarning('property "queries" must be an Array');
+  } else {
+    const allQueriesIds = [];
+    queries.forEach((query, index) => {
```

Comment:
I'm a bit confused : is `queries == null`, the `else` block will be executed and it could provoke a crash on `forEach` ()the same question about `categories` on lines 123-126). Is it a real hypothesis or `queries` cannot be `null` ? :slightly_smiling_face:

Reply:
That's correct, it shows that I should probably write a unit test for this function too :smile: